src/java.base/share/classes/java/lang/VirtualThread.java

```diff
@@ -270,11 +270,10 @@ private void lazySubmitRunContinuation() {
     @ChangesCurrentThread
     private void run(Runnable task) {
         assert state == RUNNING;
-        boolean notifyJvmti = notifyJvmtiEvents;
```

Comment:
Don't we have same issue in yieldContinuation() method? (line 396)

Reply:
Fixed the `yieldContinuation()` method.
There is also `switchToCarrierThread()` method that returns the `notifyJvmtiEvents` value.
It seems to be an optimization. I'm not sure yet, if we need to fix these places as well.

Reply:
> There is also `switchToCarrierThread()` method that returns the `notifyJvmtiEvents` value.
> It seems to be an optimization. I'm not sure yet, if we need to fix these places as well.

It was to ensure that hide(true) and hide(false) are balanced. If it were to re-poll notifyJvmtiEvents and a JVMTI agent enables the capability while a thread in doing a temporary transition then you may get a hide(false) without the corresponding hide(true).

Reply:
Okay, I see it now.